### PR TITLE
[OPPRO-326] Add byte array bool validation

### PR DESCRIPTION
**Pull request by JkSelf**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by JkSelf** [comment]
@rui-mo 
 Please help to review. Thanks for your help!

File: gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala [hunk 1]
```diff
@@ -75,7 +76,25 @@ trait BasicScanExecTransformer extends TransformSupport {
...
+      case bool: BooleanType =>
+    }.nonEmpty
+  }
+
   override def doValidate(): Boolean = {
+    // TODO need also check orc file format
+    ConverterUtils.getFileFormat(this) match {
+      case ParquetReadFormat =>
+        if (BackendsApiManager.getBackendName.equals("velox") &&
+          unsupportedDataType()) {
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:92]: Could you explain why this part of logic is not added into native validator?

Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:92]: And maybe `GlutenConfig.getConf.isVeloxBackend` can work here.

Comment (JkSelf) [reply gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:92]: It is more easier to do get the data type in java side than native side. 
`GlutenConfig.getConf.isVeloxBackend` is removed in latest code.

Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:92]: It looks good to me as a temporary solution, but in the long term, native validator should still be responsible to tell Gluten whether the backend supports the computing.

**Review by rui-mo — APPROVED** [review]

File: gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala (hunk 1 above)
Comment (JkSelf) [reply gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:92]: Ok. I will add a TODO here.

**Pull request merged**